cpp/src/reductions/scan.cu

```diff
@@ -213,15 +213,18 @@ struct ScanDispatcher {
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
-    std::unique_ptr<column> output;
-    if (inclusive == scan_type::INCLUSIVE)
-      output = inclusive_scan<T>(input, null_handling, stream, mr);
-    else
-      output = exclusive_scan<T>(input, null_handling, stream, mr);
+    auto output = [&] {
+      using Type = device_storage_type_t<T>;
+      return inclusive == scan_type::INCLUSIVE
+               ? inclusive_scan<Type>(input, null_handling, stream, mr)
+               : exclusive_scan<Type>(input, null_handling, stream, mr);
+    }();
+
     if (null_handling == null_policy::EXCLUDE) {
       CUDF_EXPECTS(input.null_count() == output->null_count(),
                    "Input / output column null count mismatch");
     }
```

Comment:
Unrelated to your PR, but this is a weird error check. This would only fire if the implementation of `inclusive/exclusive_scan` was incorrect. Usually we should be error checking against faulty user inputs. If anything, this is more of an `assert`, but should probably be removed all together.

Reply:
Yea, i had a similar thought. I wasn't used to seeing "post-condition" checks.